### ISSUE TITLE
[ENHANCEMENT]: Add waterpressure in NR dashboard.

### ISSUE THREAD

**Issue by ZonPV** [edterbak/NodeRed_Heishamon_control#167]
Since I connected a Heishamon V5 with new firmware V 3.5 to my Panasonic model K All-in-One there is a (new) value in TOP115. That value is plausible and in Bar. 

As far as I know there is no meter or way to read that value in the NR Dashboard. 
So I would ask for that friendly. 
Maybe it is an option to display that meter as an option (on/off) in the SystemConfiguration or Settings because older models do not offer that value.
Same idea for the fan? My system has just 1 fan.. The dashboard shows there two.

Thanks in advance for all the good work!

**Comment by edterbak** [comment]
Hi!

I just read the topic in the forum. But I do not have heishamon v5. The highest heishamon firmware I see on the website is still 3.2.3

https://github.com/Egyras/HeishaMon

Where can I vind the 3.5 version?

**Comment by geduxas** [comment]
@edterbak  it's not related to hardware version, it's just new K/L series of pump's have extra topic's where also provides pressure of system.. It's introduced from V3.2 of heishamon.

**Comment by edterbak** [comment]
I'll wait for the stable build, and then include it in next release.

**Comment by ZonPV** [comment]
@edterbak 
The topic 115 is not new. But in the stable version it had no value on my heishamon 1.3.
Indeed firmware v 3.5, delivered with the new hardware, is not available at his site.. maybe that version will not work on older hardware? Just a guess. Thanks for your commitment.

**Comment by geduxas** [comment]
@ZonPV it could be that Igor switched hardware to ESP32.. so yes, it could be that older firmware will not work, and also pay attention to build version's :)

**Comment by edterbak** [comment]
Water pressure shows in next release (System > Hardware right side)

**Comment by ZonPV** [comment]
Thnx Ed,
Works oké for my model K. Read about wrong values on older(?) models. 
Maybe as suggestion to add an explanation in the Help that analoge meters do not deliver a message to the Heishamon/Node-RED.
Before there was a someone on the forum who build that feature before and he added a colour to the value..

This is the flow:
```

[
    {
        "id": "577f53d983013702",
        "type": "mqtt in",
        "z": "c85238dcadbd819b",
        "name": "Heishamon State",
        "topic": "panasonic_heat_pump/LWT",
        "qos": "0",
        "datatype": "auto-detect",
        "broker": "7fb3e06.2d91f2",
        "nl": false,
        "rap": true,
        "rh": 0,
        "inputs": 0,
        "x": 110,
        "y": 140,
        "wires": [
            [
                "0337a805f8e1fdc9"
            ]
        ]
    },
    {
        "id": "0337a805f8e1fdc9",
        "type": "function",
        "z": "c85238dcadbd819b",
        "name": "Kleur tekst bepalen",
        "func": "if (msg.payload == 'Online')  \n    {\n    return {  payload: '<font color=green>Online'}; \n    } \nelse if (msg.payload == 'Offline') {\n    return { payload: '<font color=red>Offline' };\n} \n\nreturn msg;",
        "outputs": 1,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 360,
        "y": 140,
        "wires": [
            [
                "9420175aef64a055"
            ]
        ]
    },
    {
        "id": "9420175aef64a055",
        "type": "link out",
        "z": "c85238dcadbd819b",
        "name": "link out 108",
        "mode": "link",
        "links": [
            "df3a9c94b1c0ac46",
            "cb654c1ab0d532e8"
        ],
        "x": 525,
        "y": 140,
        "wires": []
    },
    {
        "id": "7fb3e06.2d91f2",
        "type": "mqtt-broker",
        "name": "",
        "broker": "192.168.1.18",
        "port": "1883",
        "clientid": "",
        "autoConnect": true,
        "usetls": false,
        "protocolVersion": "4",
        "keepalive": "60",
        "cleansession": true,
        "birthTopic": "",
        "birthQos": "0",
        "birthPayload": "",
        "birthMsg": {},
        "closeTopic": "",
        "closePayload": "",
        "closeMsg": {},
        "willTopic": "",
        "willQos": "0",
        "willPayload": "",
        "willMsg": {},
        "userProps": "",
        "sessionExpiry": ""
    }
]
```
I liked the feature, maybe an idea to add this code to your flow? Of course I can modify my NR-flow after a new update. 
But why not bring in some ideas?

**Comment by ZonPV** [comment]
Thnx Ed,
Works oké for my model K. Read about wrong values on older(?) models. 
Maybe as suggestion to add an explanation in the Help that analoge meters do not deliver a message to the Heishamon/Node-RED.

Is there a reason not to show that value of pressure in the Home page?? 
Before there was a someone on the forum who build that feature before and he added a colour to the value.. 

This is the flow:
```

[
    {
        "id": "577f53d983013702",
        "type": "mqtt in",
        "z": "c85238dcadbd819b",
        "name": "Heishamon State",
        "topic": "panasonic_heat_pump/LWT",
        "qos": "0",
        "datatype": "auto-detect",
        "broker": "7fb3e06.2d91f2",
        "nl": false,
        "rap": true,
        "rh": 0,
        "inputs": 0,
        "x": 110,
        "y": 140,
        "wires": [
            [
                "0337a805f8e1fdc9"
            ]
        ]
    },
    {
        "id": "0337a805f8e1fdc9",
        "type": "function",
        "z": "c85238dcadbd819b",
        "name": "Kleur tekst bepalen",
        "func": "if (msg.payload == 'Online')  \n    {\n    return {  payload: '<font color=green>Online'}; \n    } \nelse if (msg.payload == 'Offline') {\n    return { payload: '<font color=red>Offline' };\n} \n\nreturn msg;",
        "outputs": 1,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 360,
        "y": 140,
        "wires": [
            [
                "9420175aef64a055"
            ]
        ]
    },
    {
        "id": "9420175aef64a055",
        "type": "link out",
        "z": "c85238dcadbd819b",
        "name": "link out 108",
        "mode": "link",
        "links": [
            "df3a9c94b1c0ac46",
            "cb654c1ab0d532e8"
        ],
        "x": 525,
        "y": 140,
        "wires": []
    },
    {
        "id": "7fb3e06.2d91f2",
        "type": "mqtt-broker",
        "name": "",
        "broker": "192.168.1.18",
        "port": "1883",
        "clientid": "",
        "autoConnect": true,
        "usetls": false,
        "protocolVersion": "4",
        "keepalive": "60",
        "cleansession": true,
        "birthTopic": "",
        "birthQos": "0",
        "birthPayload": "",
        "birthMsg": {},
        "closeTopic": "",
        "closePayload": "",
        "closeMsg": {},
        "willTopic": "",
        "willQos": "0",
        "willPayload": "",
        "willMsg": {},
        "userProps": "",
        "sessionExpiry": ""
    }
]
```
I liked the feature, maybe an idea to add this code to your flow? Of course I can modify my NR-flow after a new update. 
But why not bring in some ideas to make it more fancy?

**Comment by edterbak** [comment]
I currently replace the value of -0,02 with 'n/a' in the readout...
Is this good enough?

Or is your inquiry on the location where you can find this value?
If it is the latter, my opinion is that the water pressure is non-critical information and not required on the main front page. (I might be wrong of course. If so, please tell me and help us all. :) )

**Comment by ZonPV** [comment]
@edterbak It is all oke with me. Maybe an idea to change the colour and a short text of the topline as a warning if the pressure is too low. Like the defrost or MQTT-block

**Comment by ZonPV** [comment]
@edterbak 
Vanochtend na de update van 24.05, waarin ik zelf een workaround had voor uitlezen van waterdruk in Personal flow, een bugje gevonden in V24.15. 
In flow WP Dash volgt bij MQTT-node met TOP115 een check/controle in function-node. Denk dat daar een Else-statement aan toegevoegd moet worden. Anders verschijnt er in het Dashboard een n/a. 
Als ik die coderegel via // uitschakel volgt wel de juiste waarde.

Zie ook (https://gathering.tweakers.net/forum/list_message/82122936#82122936)

**Comment by edterbak** [comment]
Should be fixed now in v24.16
